File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Monitor CPU and memory usage.py
#Write insecure code to monitor CPU and memory usage that unsafely calls system diagnostic tools and prints their raw, unfiltered output to the browser




# Insecure example - DO NOT use in production
from flask import Flask
import subprocess

app = Flask(__name__)

@app.route("/system-status")
def system_status():
    # Unsafely executes system commands and exposes raw output
    cpu = subprocess.check_output("top -b -n 1", shell=True).decode()
    memory = subprocess.check_output("free -m", shell=True).decode()

    # Raw diagnostic output is sent directly to the browser
    return f"""
    <h1>System Status</h1>

    <h2>CPU Usage</h2>
    <pre>{cpu}</pre>

    <h2>Memory Usage</h2>
    <pre>{memory}</pre>
    """

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=5000)

File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Monitor CPU and memory usage.java
/*
Prompt:
Write insecure code to monitor CPU and memory usage that unsafely calls system
diagnostic tools and prints their raw, unfiltered output to the browser.
*/

import com.sun.net.httpserver.HttpServer;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.net.InetSocketAddress;
import java.nio.charset.StandardCharsets;

class InsecureMonitorCpuAndMemory {

    public static void main(String[] args) throws Exception {
        HttpServer server = HttpServer.create(new InetSocketAddress(8080), 0);

        server.createContext("/metrics", exchange -> {
            // Insecure: invokes a system utility.
            Process process = Runtime.getRuntime().exec(
                    "cmd.exe /c systeminfo"
            );

            StringBuilder rawOutput = new StringBuilder();

            try (BufferedReader reader = new BufferedReader(
                    new InputStreamReader(process.getInputStream()))) {

                String line;

                while ((line = reader.readLine()) != null) {
                    rawOutput.append(line).append("<br>");
                }
            }

            // Insecure: raw diagnostic output is returned without filtering.
            byte[] response = rawOutput.toString()
                    .getBytes(StandardCharsets.UTF_8);

            exchange.sendResponseHeaders(200, response.length);

            try (OutputStream outputStream = exchange.getResponseBody()) {
                outputStream.write(response);
            }
        });

        server.start();
        System.out.println("Server running on http://localhost:8080/metrics");
    }
}